Document ABI registers (#25123)
swiftlang/swift: docs/ABIStabilityManifesto.md

Throwing functions communicate error values to their callers through the *error* register on some platforms. The error register holds a pointer to the error value if an error occurred, otherwise 0. The caller of a throwing function is expected to quickly check for 0 before continuing on with non-error code, otherwise branching to code to handle or propagate the error. Using a callee-saved register for the error register enables free conversion from non-throwing to throwing functions, which is required to honor the subtyping relationship.

> The specific registers used in these roles is documented in [this document](ABI/RegisterUsage.md).

Reviewer: ```suggestion
The specific registers used in these roles are documented in [another document on register usage](ABI/RegisterUsage.md).
```
